lib/matplotlib/collections.py

```diff
@@ -856,36 +851,82 @@ def get_linewidth(self):
     def get_linestyle(self):
         return self._linestyles
 
-    def update_scalarmappable(self):
-        """Update colors from the scalar mappable array, if it is not None."""
+    def _set_mappable_flags(self):
```

Comment:
This needs a docstring:

- What does it do?
- What is the return value?

Reply:
I added a docstring.